src/components/BMDashboard/UpdateConsumable/UpdateConsumable.jsx

```diff
@@ -0,0 +1,310 @@
+import './UpdateConsumable.css';
+import * as moment from 'moment';
+import { Container, FormGroup, Input, Label, Form, Col, Button } from 'reactstrap';
+import { useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { useEffect } from 'react';
+import { toast } from 'react-toastify';
+import { postConsumableUpdate } from '../../../actions/bmdashboard/consumableActions';
+
+function UpdateConsumable({ record, setModal }) {
+  const dispatch = useDispatch();
+  const postConsumableUpdateResult = useSelector(state => state.bmConsumables.updateConsumables);
+  const { purchaseRecord, stockAvailable, updateRecord: _, ...rest } = record;
+  const recordInitialState = {
+    date: moment(new Date()).format('YYYY-MM-DD'),
+    quantityUsed: '0',
+    quantityWasted: '0',
+    QtyUsedLogUnit: 'unit',
+    QtyWastedLogUnit: 'unit',
+    consumable: rest,
+    newAvailable: undefined,
+  };
+  const validationsInitialState = {
+    quantityUsed: '',
+    quantityWasted: '',
+    quantityTogether: '',
+  };
+  const [updateRecord, setUpdateRecord] = useState(recordInitialState);
+  const [validations, setValidations] = useState(validationsInitialState);
+  const [availableCount, setAvailableCount] = useState(undefined);
+  const [changeOccured, setChangeOccured] = useState(false);
+
+  useEffect(() => {
+    setUpdateRecord({ ...recordInitialState });
+    setValidations({ ...validationsInitialState });
+  }, []);
+
+  useEffect(() => {
+    if (postConsumableUpdateResult.loading === false && postConsumableUpdateResult.error === true) {
+      toast.error(`${postConsumableUpdateResult.result}`);
+      setModal(false);
+    } else if (
+      postConsumableUpdateResult.loading === false &&
+      postConsumableUpdateResult.result !== null
+    ) {
+      toast.success(`Updated ${record?.itemType?.name} successfully`);
+      setModal(false);
+    }
+  }, [postConsumableUpdateResult]);
+
+  useEffect(() => {
+    const qtyUsedFloat = parseFloat(updateRecord.quantityUsed);
+    const qtyWastedFloat = parseFloat(updateRecord.quantityWasted);
+    if (qtyUsedFloat || qtyWastedFloat) {
+      setChangeOccured(true);
+    } else {
+      setChangeOccured(false);
+    }
+
+    // eslint-disable-next-line no-use-before-define
+    validate(
+      updateRecord.quantityUsed,
+      updateRecord.quantityWasted,
+      updateRecord.QtyUsedLogUnit,
+      updateRecord.QtyWastedLogUnit,
+    );
+  }, [updateRecord]);
+
+  const validate = (_qtyUsed, _qtyWasted, QtyUsedLogUnit, QtyWastedLogUnit) => {
```

Comment:
These variables `QtyUsedLogUnit`, `QtyWastedLogUnit` doesn't seem to conform with common naming convention in JS.

Reply:
@kylepham  you have a point. These variable names mostly carried over from similar components created by other people. I made some changes to resolve merge conflicts and also changed these variable names on the front end and the backend. Please review again when you have a chance and give me your approval if everything works